### PR TITLE
Add RegexLiteral definition

### LINKED ISSUE #27: Standardize new RegExp literal AST format
In Esprima, `/foo/gi` now gives:

```js
{
  type: "Literal",
  value: /foo/gi,
  regex: {
    pattern: "foo",
    flags: "gi"
  }
}
```

The advantage of having the separate `regex` property is that even in environments that don’t support a given flag (e.g. the `u` flag) the regex can still be represented in the AST. The `value` property is `null` in such a case.

See <https://code.google.com/p/esprima/issues/detail?id=557#c4>.

cc @sebmck @michaelficarra

### PR DESCRIPTION
**Pull request by sebmck**
Closes #27. Both Esprima and Acorn follow this representation.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -603,6 +603,19 @@ interface Literal <: Node, Expression {
...
+```js
+interface RegexLiteral <: Literal {
+  regex: {
+    pattern: string;
+    flags: string;
+  };
+}
+```
+
+The `regex` property allows regexes to be represented in environments that don’t
```
Comment (mathiasbynens) [inline spec.md]: Should there be a note saying that the `value` property can be `null` in such cases?

Comment (RReverser) [inline spec.md]: Yeah, I guess so.

Comment (impinball) [inline spec.md]: This is really helpful.

Comment (sebmck) [inline spec.md]: Oh yeah, knew I forgot something.

Comment (sebmck) [inline spec.md]: Updated.

**Comment by RReverser** [comment]
I guess we can merge this as it reflects already working implementation. Thoughts?

**Comment by sebmck** [comment]
Merging this since it reflects the behaviour of Espree, Esprima and Acorn and there don't appear to be any objections.

**Comment by mikesherov** [comment]
Yeah, just as a note. Consider anything that is in all 3 as "merge at will". Thanks for moving this forward. :)

**Pull request merged**